lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+        let container                  = DomContainer::from_element(element);
+        let front_dom    : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_dom     : HtmlElement = dyn_into(create_element("div")?)?;
+        let front_camera : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_camera  : HtmlElement = dyn_into(create_element("div")?)?;
+
+        front_dom.set_property_or_panic("position", "absolute");
+        front_dom.set_property_or_panic("top", "0px");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("width", "100%");
+        front_dom.set_property_or_panic("height", "100%");
+        front_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("position", "absolute");
+        back_dom.set_property_or_panic("top", "0px");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("width", "100%");
+        back_dom.set_property_or_panic("height", "100%");
+        back_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("z-index", "-1");
+        front_camera.set_property_or_panic("width", "100%");
+        front_camera.set_property_or_panic("height", "100%");
+        front_camera.set_property_or_panic("transform-style", "preserve-3d");
+        back_camera.set_property_or_panic("width", "100%");
+        back_camera.set_property_or_panic("height", "100%");
+        back_camera.set_property_or_panic("transform-style", "preserve-3d");
+
+        container.dom.append_or_panic(&front_dom);
+        container.dom.append_or_panic(&back_dom);
+        front_dom.append_or_panic(&front_camera);
+        back_dom.append_or_panic(&back_camera);
+
+        let data             = Css3dRendererData::new(front_dom, back_dom, front_camera, back_camera);
+        let data             = Rc::new(data);
+        let mut renderer     = Self {container,data,logger};
+
+        renderer.init_listeners();
+        Ok(renderer)
+    }
+
+    /// Creates a Css3dRenderer.
+    pub fn new<L:Into<Logger>>(logger:L, dom_id: &str) -> Result<Self> {
+        Self::from_element(logger,dyn_into(get_element_by_id(dom_id)?)?)
+    }
+
+    pub(crate) fn logger(&self) -> Logger {
+        self.logger.clone()
+    }
+
+    fn init_listeners(&mut self) {
+        let dimensions = self.dimensions();
+        self.set_dimensions(dimensions);
+        let data = self.data.clone();
+        self.add_resize_callback(move |dimensions:&Vector2<f32>| {
+            data.set_dimensions(*dimensions);
+        });
+    }
+
+    /// Creates a new instance of Css3dObject and adds it to parent.
+    pub(super) fn new_instance
+    <S:AsRef<str>>(&self, dom_name:S, parent:DisplayObjectData) -> Result<Css3dObject> {
```

Comment:
Here and other places `AsRef<str>` -> `Str`.

Reply:
I changed every reference in the project. Except the ones in `build-utilities`.